packages/forms-web-app/src/dynamic-forms/dynamic-components/multi-file-upload/question.js

```diff
@@ -10,6 +32,146 @@ class MultiFileUploadQuestion extends Question {
+	 * @returns
+	 */
+	saveAction = async (req, res, journey, sectionObj, journeyResponse) => {
+		try {
+			const { body } = req;
+			// const { errors = {}, errorSummary = [] } = body;
+			const encodedReferenceId = encodeURIComponent(journeyResponse.referenceId);
+
+			// set answer on response
+			let responseToSave = {
+				answers: {
+					[this.fieldName]: {
+						uploadedFiles: []
+					}
+				}
+			};
+
+			// Ensure we are always working with an array. Single files would otherwise be an object.
+			if (!Array.isArray(req.files[this.fieldName])) {
+				req.files[this.fieldName] = [req.files[this.fieldName]];
+			}
+
+			// remove files from response
+			if ('removedFiles' in body) {
+				const removedFiles = JSON.parse(body.removedFiles);
+
+				journeyResponse.answers[this.fieldName].uploadedFiles = removeFiles(
+					journeyResponse.answers[this.fieldName].uploadedFiles,
+					removedFiles
+				);
+			}
+
+			// save files to blob storage
+			const uploadedFiles = await this.#saveFilesToBlobStorage(req, journeyResponse);
+
+			// add saved docs to response
+			responseToSave.answers[this.fieldName].uploadedFiles = [];
+
+			if (journeyResponse?.answers[this.fieldName]?.uploadedFiles) {
+				responseToSave.answers[this.fieldName].uploadedFiles.push(
+					...journeyResponse.answers[this.fieldName].uploadedFiles
+				);
+			}
+
+			if (uploadedFiles) {
+				responseToSave.answers[this.fieldName].uploadedFiles.push(...uploadedFiles);
+			}
+
+			// save answer to database
+			await patchQuestionResponse(journeyResponse.journeyId, encodedReferenceId, responseToSave);
+
+			// todo: validation after saving required?
+			// if (Object.keys(errors).length > 0) {
+			// 	const answer = journeyResponse.answers[this.fieldName].uploadedFiles;
+			// 	let viewModel = journey.getQuestionViewModel(sectionObj, this, answer);
+			// 	viewModel.errors = errors;
+			// 	// multi-file upload validation would otherwise map these errors individual to e.g.
+			// 	// `#files.supporting-documents[3]` which does not meet the gov uk presentation requirements.
+			// 	viewModel.errorSummary = errorSummary.map((error) => ({
+			// 		...error,
+			// 		href: '#'
+			// 	}));
+			// 	return this.#showError(res, viewModel);
+			// }
+
+			// map the response back to journeyResponse
+			journeyResponse.answers[this.fieldName] = responseToSave.answers[this.fieldName];
+
+			// update journey based on response
+			const updatedJourney = new journey.constructor(journeyResponse);
+
+			// this is the `name` of the 'upload' button in the template.
+			// when is this used?
+			if (body['upload-and-remain-on-page']) {
+				return res.redirect(
+					updatedJourney.getCurrentQuestionUrl(sectionObj.segment, this.fieldName)
+				);
+			}
+
+			return res.redirect(
+				updatedJourney.getNextQuestionUrl(sectionObj.segment, this.fieldName, false)
+			);
+		} catch (err) {
+			logger.error(err);
+
+			return this.#showError(res, journey, sectionObj, {
```

Comment:
I'm guessing this is for a scenario where validation ext passes but there is some other issue that prevents the files being saved?

Reply:
Potentially yes, have been unable to trigger it, but can leave in for now

Reply:
I think it's a good thing to get a test against. Could we trigger it in a unit test by mocking out one of the downstream dependencies to throw an error?